drivers/rilmodem/gprs-context.c

```diff
@@ -534,9 +544,27 @@ static void ril_gprs_context_remove(struct ofono_gprs_context *gc)
 	DBG("*gc: %p", gc);
 
 	if (gcd->state != STATE_IDLE) {
-		ril_gprs_context_detach_shutdown(gc, 0);
+		struct req_deactivate_data_call request;
+		struct parcel rilp;
+		struct ofono_error error;
+
+		request.cid = gcd->active_rild_cid;
+		request.reason = RIL_DEACTIVATE_DATA_CALL_NO_REASON;
+		g_ril_request_deactivate_data_call(gcd->ril, &request,
+								&rilp, &error);
+
+		g_ril_send(gcd->ril, RIL_REQUEST_DEACTIVATE_DATA_CALL,
+						&rilp, NULL, NULL, NULL);
+	}
+
```

Comment:
The replacement of the ril_gprs_context_detach_shutdown() call with an inline DEACTIVATE_DATA_CALL changes the behavior, removing:

 - a state check to see determine if the detach_shutdown could be ignored

 - unregistering of the ril listener ( seems most of our other atoms don't do this )

- the deactivate data call retry logic

Maybe these were related to the crash, but it's not obvious at first glance.